core/lib/brightcontent/base_controller_ext/pagination.rb

```diff
@@ -5,21 +5,61 @@ module BaseControllerExt
     module Pagination
       extend ActiveSupport::Concern
 
+      included do
+        helper_method :items_per_page
+        helper_method :current_page
+        helper_method :per_page_sizes
+        helper_method :corrected_page
+      end
+
       module ClassMethods
         def per_page_count
-          @per_page_count ||= 30
+          @per_page_count =
+            @per_page_sizes.try(:any?) ? @per_page_sizes[0] : (@per_page_count || 30)
         end
 
         def per_page(number)
           @per_page_count = number
         end
+
+        def page_size_options(sizes)
+          return unless sizes.is_a? Array
+          @per_page_sizes =
+            sizes.select { |size| size.is_a?(Integer) && size > 0 }
+        end
+
+        def per_page_sizes
```

Comment:
There's a slight inconsistency in naming here: `per_page_count` vs `per_page_sizes` vs `page_size_options`. (This was there previously as well with `per_page_count` the instance variable and `per_page` the class method.)

Perhaps we should use `page_size` and `page_size_options` throughout?